sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SdkContextKeys.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace
+
+import cats.effect.SyncIO
+import org.typelevel.otel4s.sdk.context.Context
+import org.typelevel.otel4s.trace.SpanContext
+
+/** The keys that can be used by a third-party integrations.
+  */
+object SdkContextKeys {
+
+  /** The [[org.typelevel.otel4s.trace.SpanContext SpanContext]] is stored under
+    * this key in the [[org.typelevel.otel4s.sdk.context.Context Context]].
+    *
+    * To retrieve the context use:
+    * {{{
+    *   val context: Context = ???
+    *   val spanContext: Option[SpanContext] = context.get(SdkContextKeys.SpanContextKey)
+    * }}}
+    */
+  val SpanContextKey: Context.Key[SpanContext] =
```

Comment:
We need a public static instance of the key that propagators can use. 
For example, [W3CTraceContextPropagator](https://github.com/typelevel/otel4s/blob/b19605aff6272c1cfb75c0b53b4e3dd6dbcb07f6/sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/propagation/W3CTraceContextPropagator.scala#L62) relies on this key.

I assume, we will need to rewrite W3C propagator in a similar manner you reimplemented [PassThroughPropagator](https://github.com/typelevel/otel4s/blob/48d4c6adfb4fd7a7cf3d49cf70dfe8a851abb924/core/common/src/main/scala/org/typelevel/otel4s/context/propagation/PassThroughPropagator.scala).

So it will be 
```scala
class W3CTraceContextPropagator[Ctx, K[X] <: Key[X]](
  spanContextKey: K[SpanContext]
)(implicit c: Contextual.Keyed[Ctx, K]) extends TextMapPropagator[Ctx] {
  ....
}
```

Even then, we will need to pass a specific key to the constructor:
```scala
new W3CTraceContextPropagator(SdkContextKeys.SpanContext)
```